torch/_inductor/codegen/halide.py

```diff
@@ -921,44 +921,92 @@ def halide_kernel_meta(self) -> HalideMeta:
         for _, arg in self.halide_argdefs():
             if isinstance(arg, SizeArg):
                 shape = None
+                stride = None
+                offset = None
                 dtype = "long"
             else:
                 if arg.name in self.store_buffer_dimensions and "out" in arg.name:
                     shape = [
                         cexpr(self.rename_indexing(x))
                         for x in self.store_buffer_dimensions[arg.name]
                     ]
-                    assert shape
+                    stride = []
+                    prod = 1
+                    for x in self.store_buffer_dimensions[arg.name]:
+                        stride.append(cexpr(prod))
```

Comment:
Wondering why halide put the fastest moving dimension at the beginning rather than in the end (like pytorch).

So halide can only support contiguous tensors?

Reply:
This is the convention in image processing -- the opposite of PyTorch.

Reply:
I see. GPU layout in triton also picks this kind of order (fastest moving dimension first).